flytekit/annotated/interface.py

```diff
@@ -260,3 +260,14 @@ def t(a: int, b: str) -> None: ...
 
     return_map = OrderedDict(zip(list(output_name_generator(len(return_types))), return_types))
     return return_map
+
+
+# Doesn't really belong here, not sure where else to put it
+class ControlPlaneSettings(object):
```

Comment:
should this not be in the context? as one more layer?

Reply:
one more layer?  you mean like an object inside the CompilationState object?  I think that would make the most sense, a registration settings object or something...

Reply:
yup, like in the compilation state. I thought we had discussed this. This would also make it possible to serialize with different compilation states?

Reply:
doing this half-way for now.  moved it into FlyteContext but leaving it out of compilation for now.  Couldn't work out how to mesh compilation and registration.  Can revisit after tomorrow.